op-service/sources/receipts.go

```diff
@@ -14,7 +14,7 @@ type ReceiptsProvider interface {
 	// FetchReceipts returns a block info and all of the receipts associated with transactions in the block.
 	// It verifies the receipt hash in the block header against the receipt hash of the fetched receipts
 	// to ensure that the execution engine did not fail to return any receipts.
-	FetchReceipts(ctx context.Context, block eth.BlockID, txHashes []common.Hash) (types.Receipts, error)
+	FetchReceipts(ctx context.Context, block eth.BlockID, txHashes []common.Hash, bInfo eth.BlockInfo) (types.Receipts, error)
```

Comment:
what about passing in the receipt hash instead of the info?

Reply:
if we keep passing `BlockInfo` we can remove the `BlockID` and build it from the `BlockInfo` (`eth.ToBlockID(info)`).